openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -578,19 +583,31 @@ def post(self, request, course_key_string, *args, **kwargs):
 
         data = dict(request.data)
         first_section = data.pop('first_section', None)
+        first_discussion = data.pop('first_discussion', None)
         if data:
             return Response(status=400)  # there were parameters we didn't recognize
 
-        enrollment = CourseEnrollment.get_enrollment(request.user, course_key)
-        if not enrollment:
-            return Response(status=404)
-
+        object_created = False
+        user_celebration = False
+        enrollment_celebration = False
         defaults = {}
         if first_section is not None:
+            enrollment = CourseEnrollment.get_enrollment(request.user, course_key)
+            if not enrollment:
+                return Response(status=404)
             defaults['celebrate_first_section'] = first_section
+            enrollment_celebration = True
+
+        if first_discussion is not None:
+            defaults['celebrate_first_discussion_post'] = first_discussion
+            user_celebration = True
 
         if defaults:
-            _, created = CourseEnrollmentCelebration.objects.update_or_create(enrollment=enrollment, defaults=defaults)
-            return Response(status=201 if created else 200)
-        else:
-            return Response(status=200)  # just silently allow it
+            created = False
+            if enrollment_celebration:
+                # _, created = CourseEnrollmentCelebration.objects.update_or_create(enrollment=enrollment, defaults=defaults)
+                object_created = object_created or created
+            if user_celebration:
+                # _, created = UserCelebration.objects.update_or_create(user=request.user, defaults=defaults)
+                object_created = object_created or created
+        return Response(status=201 if object_created else 200)
```

Comment:
Realized we don't need the if/else because the else is covered here anyway by doing the 200.